File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    for i in range(1,n+1):
        temp = i
        if temp == n:
            answer += 1
            break
        for j in range(i+1,n+1):
            temp += j
            if temp == n:
                answer += 1
            elif temp > n:
                break
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    int left = 0;
    int right = 1;
    int sum = left + right;
    
    while (true) {
        if (sum < n) {
            if (right > n) {
                break;
            }
            right++;
            sum += right;
        } else if (sum > n) {
            sum -= left;
            left++;
        } else {
            answer++;
            right++;
            sum += right;
        }
    }
    return answer;
}